package/broom.py

```diff
@@ -0,0 +1,310 @@
+    plays = set()
+    for card in player.hand:
+      if (len(table_cards) > 0):
+        combo_cards = set(table_cards)
+        escombinations = get_escombinations(combo_cards,card)
+        for combo in escombinations:
+          plays.add(combo)
+      else:
+        plays.add(tuple(player.hand))
+    return plays
+
+  def apply_play(self,play, player):
+    # validate(play)
+    playable = self.valid_plays(player, self.table_cards)
+    scored = False
+    t_cards = self.table_cards
+    #p_cards = play.pop()
+    p_cards = play
+    if p_cards in playable:
+      if isinstance(p_cards,str):
+        card_values = [self.deck.card_store[p_cards].face]
+      else:
+        card_values = [self.deck.card_store[c].face for c in p_cards ]
+      # assign card owners
+      s = sum(card_values)
+      if ( s == 15):
+        scored = True
+        for card in p_cards:
+          self.set_card_owner(card, player.name)
+          if card in self.table_cards:
+            self.table_cards.discard(card)
+      else:
+        if isinstance(p_cards, str):
+          self.table_cards.update({p_cards})
+        else:
+          self.table_cards.update(p_cards)
+    
+    for card in p_cards:
+      if card in player.hand:
+        player.hand.discard(card)
+    if not self.table_cards:
+      player.award_point() #escoba
+      print(f"{player.name} Escoba!")
+      self.print_score()
+    return scored
+
+  def apply_score(self):
+    p1_total = set()
+    p1_oros = set()
+    p1_sevens = set()
+
+    p2_total = set()
+    p2_sevens = set()
+    p2_oros = set()
+
+    for card_id, card in self.deck.cards().items():
+      if (card.owner == self.pl1.name):
+        p1_total.add(card_id)
+        if card.suit == 'O': p1_oros.add(card_id) 
+        if card.face == 7: p1_sevens.add(card_id) 
+      else:
+        p2_total.add(card_id)
+        if card.suit == 'O': p2_oros.add(card_id)
+        if card.face == 7: p2_sevens.add(card_id)
+
+      if card_id == '7O':
+        self.pl1.award_point() if card.owner == self.pl1.name else self.pl2.award_point()
+      
+    if len(p1_total) > len(p2_total):
+      self.pl1.award_point()
+    elif len(p2_total) > len(p1_total):
+      self.pl2.award_point()
+    
+    if len(p1_oros) > len(p2_oros):
+      self.pl1.award_point()
+    elif len(p2_oros) > len(p1_oros):
+      self.pl2.award_point()
+    
+    if len(p1_sevens) > len(p2_sevens):
+      self.pl1.award_point()
+    elif len(p2_sevens) > len(p1_sevens):
+      self.pl2.award_point()
+    print(f'Points:\tPL1\tPL2\nOros:\t[{len(p1_oros)}]\t[{len(p2_oros)}]\nSevens:\t[{len(p1_sevens)}]\t[{len(p2_sevens)}]\nCards:\t[{len(p1_total)}]\t[{len(p2_total)}]')
+
+  def play_first_round(self, first_player, second_player):
+    p1_cards, p2_cards ,table_cards = self.deal_start()
+    first_player.new_hand(p1_cards)
+    second_player.new_hand(p2_cards)
+    self.table_cards = table_cards
+    return self
```

Comment:
this shouldn't be needed after our discussion right?

Reply:
perhaps it should be called init game.  I left it in there because i still want to be able to instantiate the complete game object at least once in frontend.  future api calls would only transmit necessary info for that action.  i am open to doing it differently though. suggestions?